test/integration-tests/V3ToV4Migration.test.ts

```diff
@@ -945,4 +961,301 @@ describe('V3 to V4 Migration Tests:', () => {
+
+      planner.addCommand(CommandType.V4_POSITION_MANAGER_CALL, [encodedModifyLiquiditiesCallWithMint])
+
+      await executeRouter(planner, bob, router, wethContract, daiContract, usdcContract)
+
+      // increase position second
+      planner = new RoutePlanner()
+      await usdcContract.connect(bob).transfer(v4PositionManager.address, expandTo6DecimalsBN(10000))
+      await wethContract.connect(bob).transfer(v4PositionManager.address, expandTo18DecimalsBN(10))
+
+      const encodedModifyLiquiditiesCallWithIncrease = increaseSettleSweep()
+
+      planner.addCommand(CommandType.V4_POSITION_MANAGER_CALL, [encodedModifyLiquiditiesCallWithIncrease])
+
+      await executeRouter(planner, bob, router, wethContract, daiContract, usdcContract)
+
+      // bob successfully sweeped his usdc and weth from the v4 position manager
+      expect(await wethContract.balanceOf(v4PositionManager.address)).to.eq(0)
+      expect(await usdcContract.balanceOf(v4PositionManager.address)).to.eq(0)
+    })
+
+    it('migrate with minting succeeds', async () => {
+      // Bob max-approves the v3PM to access his USDC and WETH
+      await usdcContract.connect(bob).approve(v3NFTPositionManager.address, MAX_UINT)
+      await wethContract.connect(bob).approve(v3NFTPositionManager.address, MAX_UINT)
+
+      // mint the nft to bob on v3
+      const tx = await v3NFTPositionManager.mint({
+        token0: USDC.address,
+        token1: WETH.address,
+        fee: FeeAmount.LOW,
+        tickLower: 0,
+        tickUpper: 194980,
+        amount0Desired: expandTo6DecimalsBN(2500),
+        amount1Desired: expandTo18DecimalsBN(1),
+        amount0Min: 0,
+        amount1Min: 0,
+        recipient: bob.address,
+        deadline: MAX_UINT,
+      })
+
+      const receipt = await tx.wait()
+      const transferEvent = receipt.events?.find((event) => event.event === 'IncreaseLiquidity')
+      let tokenIdv3 = transferEvent?.args?.tokenId
+
+      // permit, decrease, collect, burn
+      const encodedErc721PermitCall = await permit()
+      const encodedDecreaseCall = await decreaseLiquidity()
+      // set receiver to v4posm
+      const encodedCollectCall = collect()
+      const encodedBurnCall = encodeBurn(tokenIdv3)
+
+      planner.addCommand(CommandType.V3_POSITION_MANAGER_PERMIT, [encodedErc721PermitCall])
+      planner.addCommand(CommandType.V3_POSITION_MANAGER_CALL, [encodedDecreaseCall])
+      planner.addCommand(CommandType.V3_POSITION_MANAGER_CALL, [encodedCollectCall])
+      planner.addCommand(CommandType.V3_POSITION_MANAGER_CALL, [encodedBurnCall])
+
+      let encodedModifyLiquiditiesCallWithMint = mintSettleSweep()
+
+      planner.addCommand(CommandType.V4_POSITION_MANAGER_CALL, [encodedModifyLiquiditiesCallWithMint])
+
+      await executeRouter(planner, bob, router, wethContract, daiContract, usdcContract)
+
+      // bob successfully sweeped his usdc and weth from the v4 position manager
```

Comment:
check that Bob no longer owns a position on v3, and that he now owns a position on v4

Reply:
just generally more of these checks - that position 721s burn/mint where expected, and that liquidity exists when it should